blueprints/app/index.js

```diff
@@ -21,6 +21,18 @@ module.exports = {
     let name = stringUtil.dasherize(rawName);
     let namespace = stringUtil.classify(rawName);
 
+    let hasOptions = !options.welcome || options.yarn;
+    let blueprintOptions = '';
+    if (hasOptions) {
+      let indent = `\n            `;
+      let outdent = `\n          `;
+
+      blueprintOptions =
+        indent +
+        [!options.welcome && '"--no-welcome"', options.yarn && '"--yarn"'].filter(Boolean).join(',\n            ') +
+        outdent;
+    }
```

Comment:
This is very similar between app and addon blueprints, and I tried to end up sharing the code but the shared version was much harder to reason about. This is specifically because the default values for `--welcome` (and therefore the flag value) **change** between the `app` and `addon` blueprint.